### PR TITLE
refactor(FOROME-1166): unify modal and panel for custom inh mode

### PR DESCRIPTION
**Pull request by Lapk1n**
https://quantori.atlassian.net/browse/FOROME-1166

### CODE REVIEW

**Comment by Lapk1n** [comment]
Screenshot 2022-06-08 at 11 41 06
Margin-bottom under the header of the modal will be fixed after this component is refactored to `Dialog`

File: src/components/custom-inheritance-mode-condition/components/custom-inheritance-mode-scenario.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import { t } from '@i18n'
+import { Select } from '@ui/select'
+import { ICustomInheritanceModeScenarioProps } from '../custom-inheritance-mode.interface'
+
+const selectOptions = ['--', '0', '0-1', '1', '1-2', '2']
+
+export const CustomInheritanceModeScenario = ({
+  problemGroups,
+  selectValues,
+  handleSetSingleScenario,
```
Comment (AlMaQntr) [inline src/components/custom-inheritance-mode-condition/components/custom-inheritance-mode-scenario.tsx]: onChangeScenario

File: src/components/custom-inheritance-mode-condition/components/custom-inheritance-mode-scenario.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+}: ICustomInheritanceModeScenarioProps): ReactElement => (
+  <>
+    <div className="text-14 leading-16px font-medium text-grey-blue mb-2.5">
+      {t('dtree.scenario')}
```
Comment (AlMaQntr) [inline src/components/custom-inheritance-mode-condition/components/custom-inheritance-mode-scenario.tsx]: Can we move strings into `inheritanceMode` namespace?

File: src/components/custom-inheritance-mode-condition/custom-inheritance-mode-condition.tsx [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    const [preparedValue, setPreparedValue] = useState<string>('')
+
+    const [scenario, setScenario] = useState<IScenario>(initialScenario || {})
+
+    const [selectValues, setSelectValues] = useState<string[]>([])
+
+    const toggleMode = (mode: ModeTypes) => {
+      setMode(currentMode => (currentMode === mode ? undefined : mode))
+
+      filterStore.setTouched(true)
```
Comment (AlMaQntr) [inline src/components/custom-inheritance-mode-condition/custom-inheritance-mode-condition.tsx]: Can we use callback prop `onTouch` and avoid to use `filterStore`

File: src/components/custom-inheritance-mode-condition/custom-inheritance-mode-condition.tsx [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+    }
+
+    useEffect(() => {
+      statFuncStore.setQuery({
+        unit: FuncStepTypesEnum.CustomInheritanceMode,
+        param: JSON.stringify({ scenario }),
+      })
+      // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (AlMaQntr) [inline src/components/custom-inheritance-mode-condition/custom-inheritance-mode-condition.tsx]: I think you can provide `statFuncStore` as dependency and remove this comment

**Review by AlMaQntr — APPROVED** [review]

File: src/components/inheritance-mode-condition/inheritance-mode-condition.tsx [hunk 5]
```diff
@@ -61,20 +61,20 @@ export const InheritanceModeCondition = observer(
...
+      onTouch && onTouch()
     }
 
     const selectAllVariants = () => {
       const allVariantsNames = filteredVariants?.map(
         ([variantName]) => variantName,
       )
       setSelectedVariants(allVariantsNames)
-      filterStore.setTouched(true)
+      onTouch && onTouch()
```
Comment (AndreyUstyumenko) [inline src/components/inheritance-mode-condition/inheritance-mode-condition.tsx]: onTouch?.()

File: src/pages/filter/dtree/components/modals/components/modal-custom-inheritance-mode/modal-custom-inheritance-mode.tsx [hunk 6]
```diff
@@ -1,100 +1,91 @@
...
+        initialMode={initialMode}
+        attributeSubKind={attributeSubKind}
+        statFuncStore={dtreeStatFuncStore}
+        controls={({ hasErrors, param, mode }) => {
+          return initialCondition ? (
+            <EditModalButtons
+              handleClose={
+                modalsVisibilityStore.closeModalCustomInheritanceMode
+              }
+              handleSaveChanges={() => {
```
Comment (AndreyUstyumenko) [inline src/pages/filter/dtree/components/modals/components/modal-custom-inheritance-mode/modal-custom-inheritance-mode.tsx]: move it to a function and wrap with useCallback

File: src/pages/filter/dtree/components/modals/components/modal-custom-inheritance-mode/modal-custom-inheritance-mode.tsx [hunk 7]
```diff
@@ -1,100 +1,91 @@
...
+          ) : (
+            <SelectModalButtons
+              currentGroup={currentStepGroups}
+              handleClose={
+                modalsVisibilityStore.closeModalCustomInheritanceMode
+              }
+              handleModals={handleModals}
+              handleModalJoin={modalsVisibilityStore.openModalJoin}
+              disabled={hasErrors}
+              handleAddAttribute={action => {
```
Comment (AndreyUstyumenko) [inline src/pages/filter/dtree/components/modals/components/modal-custom-inheritance-mode/modal-custom-inheritance-mode.tsx]: here too

File: src/pages/filter/refiner/components/middle-column/panels/function-panel/components/custom-inheritance-mode/custom-inheritance-mode.tsx [hunk 8]
```diff
@@ -1,108 +1,63 @@
...
+                onClick={clearValue}
+                className="px-5 mr-2"
+              />
+              <Button
+                text={
+                  initialCondition
+                    ? t('dtree.saveChanges')
+                    : t('dtree.addAttribute')
+                }
+                onClick={() =>
```
Comment (AndreyUstyumenko) [inline src/pages/filter/refiner/components/middle-column/panels/function-panel/components/custom-inheritance-mode/custom-inheritance-mode.tsx]: and here

File: src/pages/filter/refiner/components/right-column/func-filter/utils/getScenarioValue.ts [hunk 9]
```diff
@@ -0,0 +1,16 @@
+import { IScenario } from '@service-providers/common'
```
Comment (AndreyUstyumenko) [inline src/pages/filter/refiner/components/right-column/func-filter/utils/getScenarioValue.ts]: I thought we called files in kebab case

**Review by AndreyUstyumenko — APPROVED** [review]

File: src/pages/filter/refiner/components/right-column/func-filter/utils/getScenarioValue.ts (hunk 9 above)
Comment (Lapk1n) [reply src/pages/filter/refiner/components/right-column/func-filter/utils/getScenarioValue.ts]: sure

**Pull request merged**